src/shared/fido2-util.c

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "fido2-util.h"
+#include "fileio.h"
+#include "libfido2-util.h"
+#include "random-util.h"
+
+int fido2_generate_salt(struct iovec *ret_salt) {
+        _cleanup_(iovec_done) struct iovec salt = {};
+        int r;
+
+        r = crypto_random_bytes_allocate_iovec(FIDO2_SALT_SIZE, &salt);
+        if (r < 0)
+                return log_error_errno(r, "Failed to generate FIDO2 salt: %m");
+
+        *ret_salt = TAKE_STRUCT(salt);
+        return 0;
+}
+
+int fido2_read_salt_file(const char *filename, uint64_t offset, const char *client, const char *node, struct iovec *ret_salt) {
+        _cleanup_(iovec_done_erase) struct iovec salt = {};
+        _cleanup_free_ char *bind_name = NULL;
+        int r;
+
+        /* If we read the salt via AF_UNIX, make the client recognizable */
+        if (asprintf(&bind_name, "@%" PRIx64"/%s-fido2/%s", random_u64(), client, node) < 0)
```

Comment:
hmm, do we ever want to read anything else than the salt via fido2 in future? if the answer to that is "yes, we might", then I think we should include the word "salt" in this string, i.e. `…/%s-fido2-salt/…` or so?

Reply:
I mean, it could theoretically make sense, for example instead of specifying fido2-cid in /etc/crypttab, the cid could be read from the same socket, hence probably makes sense to future-proof.